code/modules/antagonists/horror/horror.dm

```diff
@@ -0,0 +1,855 @@
+		return
+	var/mob/living/captive_brain/CB = B.host_brain
+	var/input = stripped_input(src, "Please enter a message to tell the trapped mind.", "Message", null)
+	if(!input)
+		return
+
+	to_chat(CB, span_changeling("<i>[B.truename] says:</i> [input]"))
+
+	for(var/M in GLOB.dead_mob_list)
+		if(isobserver(M))
+			var/rendered = span_changeling("<i>Horror Communication from <b>[B.truename]</b> : [input]</i>")
+			var/link = FOLLOW_LINK(M, src)
+			to_chat(M, "[link] [rendered]")
+	to_chat(src, span_changeling("<i>[B.truename] says:</i> [input]"))
+
+/mob/living/simple_animal/horror/Life()
+	..()
+	if(horrorupgrades["regen"])
+		heal_overall_damage(5)
+
+	if(invisible) //don't regenerate chemicals when invisible
+		if(has_chemicals(5))
+			use_chemicals(5)
+			alpha = max(alpha - 100, 1)
+		else
+			to_chat(src, span_warning("You ran out of chemicals to support your invisibility."))
+			invisible = FALSE
+			Update_Invisibility_Button()
+	else
+		if(horrorupgrades["nohost_regen"])
+			regenerate_chemicals(chem_regen_rate)
+		else if(victim)
+			if(victim.stat == DEAD)
+				regenerate_chemicals(1)
+			else
+				regenerate_chemicals(chem_regen_rate)
+	alpha = min(255, alpha + 50)
+
+	if(victim)
+		if(stat != DEAD && victim.stat != DEAD)
+			heal_overall_damage(1)
+
+/mob/living/simple_animal/horror/say(message, bubble_type, var/list/spans = list(), sanitize = TRUE, datum/language/language = null, ignore_spam = FALSE, forced = null)
+	if(victim)
+		to_chat(src, span_warning("You cannot speak out loud while inside a host!"))
+		return
+	return ..()
+
+/mob/living/simple_animal/horror/emote(act, m_type = null, message = null, intentional = FALSE)
+	if(victim)
+		to_chat(src, span_warning("You cannot emote while inside a host!"))
+		return
+	return ..()
+
+/mob/living/simple_animal/horror/UnarmedAttack(atom/A)
+	if(istype(A, /obj/machinery/door/airlock))
+		visible_message(span_warning("[src] slips their tentacles into the airlock and starts prying it open!"), span_warning("You start moving onto the airlock."))
+		playsound(A, 'sound/misc/splort.ogg', 50, 1)
+		if(!do_after(src, 5 SECONDS, target = A))
+			return
+		visible_message(span_warning("[src] forces themselves through the airlock!"), span_warning("You force yourself through the airlock"))
+		forceMove(get_turf(A))
+		playsound(A, 'sound/machines/airlock_alien_prying.ogg', 50, 1)
+		return
+	if(isliving(A))
+		if(victim || A == src.mind.enslaved_to)
+			healthscan(usr, A)
+			chemscan(usr, A)
+		else
+			alpha = 255
+			if(hiding)
+				var/datum/action/innate/horror/H = has_ability(/datum/action/innate/horror/toggle_hide)
+				H.Activate()
+			if(invisible)
+				var/datum/action/innate/horror/H = has_ability(/datum/action/innate/horror/chameleon)
+				H.Activate()
+			Update_Invisibility_Button()
+			..()
+
+/mob/living/simple_animal/horror/ex_act()
+	if(victim)
+		return
+
+	..()
+
+/mob/living/simple_animal/horror/proc/infect_victim()
+	if(!can_use_ability())
+		return
+	if(victim)
```

Comment:
Move below the dead check. Not even sure you'd be able to infect_victim() while dead tbh